tester/orb.yaml

```diff
@@ -175,8 +193,21 @@ jobs:
           Name of docker-compose service to run
         type: string
     steps:
-      - setup_remote_docker:
-          version: 18.09.3
+      - when:
+          condition:
+            and: [<<parameters.creds>>, <<parameters.project>>]
+          steps:
+            - gcloud/install
+            - gcloud/auth:
+                creds: <<parameters.creds>>
+                project: <<parameters.project>>
+            - gcloud/configure-docker
```

Comment:
Hmm, I think adding a `gcloud` orb dependency here might be tangling things together too much -- if we can keep these orbs more as building blocks then full-fledged solutions, they'll be easier to reason about. Remember, too, these orbs are open source, so there's a bunch of folks using these without GCP.

Is this something we can solve client-side, eg. with the [`pre-steps`](https://circleci.com/docs/2.0/reusing-config/#using-pre-and-post-steps) feature of Circle's config? ie. in the relevant app's config, something like:
```yaml
- docker-compose:
    service_name: foo
    # etc
    pre-steps:
      - gcloud/install
      - gcloud/auth:
          creds: FOO_BAR
          project: baz-buuq
```
?

Reply:
I like the idea in general, but the implementation I can see is surprisingly coupled too, because of `gcloud/configure_docker` step:
```
  configure-docker:
    # ...
    steps:
      - when:
          condition: <<parameters.docker_layer_caching>>
          steps:
            - setup_remote_docker:
                version: 18.09.3
                docker_layer_caching: true
      - unless:
          condition: <<parameters.docker_layer_caching>>
          steps:
            - setup_remote_docker:
                version: 18.09.3
      - run: gcloud auth configure-docker --quiet
```
`setup_remote_docker` is also a part of `tester/docker-compose` job and in order to run `- run: gcloud auth configure-docker --quiet` after it we'll have to drop `setup_remote_docker` from  `tester/docker-compose` which presumably will make it unusable as a standalone orb.

It also arguably puts additional cognitive effort on people using it in the future. They have to know that `tester/docker-compose` by default can't fetch from GCR and you need to combine two orbs to achieve this. While I totally expect third-party orbs to need careful examination before they can be used, I'd prefer our homegrown orbs to just work™

Reply:
Oh shoot, does calling `setup_remote_docker` twice not work? Used to be that calls after the first were just noops, which was really useful for building up stuff like this.

> It also arguably puts additional cognitive effort on people using it in the future. They have to know that `tester/docker-compose` by default can't fetch from GCR and you need to combine two orbs to achieve this. 

I don't think that would be all that unexpected; eg. nor can you pull from any other container registry other than dockerhub (say, Amazon's ECR). I'm not all that comfortable having GCR be _special_ compared to any others, either. The pattern of using `pre-steps` to configure any additional dependencies for the orb you're working with is already fairly standard (eg. see where we inject additional dynamic libs in realtime-decoder).

Also a bit worried about how the chain of dependencies is turning out -- our `gcloud` orb depends on `docker` already for the GKE stuff, and now `tester` will depend on both (transitively)... that'll probably start making version management annoying for ourselves sooner rather than later.

> While I totally expect third-party orbs to need careful examination before they can be used, I'd prefer our homegrown orbs to just work™

Definitely the (sometimes unfortunate) tradeoff when you OSS something is that you have to think of use-cases other than your own ;P

Reply:
> Oh shoot, does calling setup_remote_docker twice not work? Used to be that calls after the first were just noops, which was really useful for building up stuff like this.

Hmm, it does not, unless there is some silly mistake I'm overlooking
https://app.circleci.com/pipelines/github/talkiq/realtime/2138/workflows/14da4f7a-1c66-4422-9f78-01aca0f9c6aa/jobs/11920

>I don't think that would be all that unexpected; eg. nor can you pull from any other container registry other than dockerhub (say, Amazon's ECR). I'm not all that comfortable having GCR be special compared to any others, either. The pattern of using pre-steps to configure any additional dependencies for the orb you're working with is already fairly standard (eg. see where we inject additional dynamic libs in realtime-decoder).

sorry, by users I meant solely members of the team. Theoretical accidental users of the orb should totally examine the orb before using it, while fellow colleagues, in my opinion, are in full right to expect our orbs to work with our image storage.
(If we use Amazon ECR somewhere and I'm just not aware of this I immediately recall this argument :D)

>Definitely the (sometimes unfortunate) tradeoff when you OSS something is that you have to think of use-cases other than your own ;P

Haha, yeah. Actually, the more I think about it the less I see the code enforced by a private company to be publicly available as OSS.

Reply:
> Hmm, it does not

Shoot. Well, that's a good reason to need to merge these anyway!

> Theoretical accidental users of the orb should totally examine the orb before using it, while fellow colleagues, in my opinion, are in full right to expect our orbs to work with our image storage.

That's a good point! Especially combined with the above causing "keeping these separate" to be more complexity to manage, I'm pretty well convinced here.